src/Stratis.Bitcoin/P2P/Peer/NetworkPeerListener.cs

```diff
@@ -1,78 +1,72 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Linq;
 using System.Threading;
-using NBitcoin.Protocol;
+using System.Threading.Tasks;
 using Stratis.Bitcoin.P2P.Protocol;
 using Stratis.Bitcoin.P2P.Protocol.Payloads;
+using Stratis.Bitcoin.Utilities;
 
 namespace Stratis.Bitcoin.P2P.Peer
 {
-    public class NetworkPeerListener : PollMessageListener<IncomingMessage>, IDisposable
+    /// <summary>
+    /// Message listener that waits until a specific payload is received and returns it to the caller.
+    /// </summary>
+    /// <typeparam name="T">Type of the messages that are being handled.</typeparam>
```

Comment:
There is no param with name T. Did you mean `IncomingMessage`?

Reply:
Thank you will fix